### PR TITLE
fix(CI): Make sure the "when unrelated" fake summaries work correctly

### PR DESCRIPTION
**Pull request by nickvergessen**
Before they executed when any file outside the defined scope matched. However that means: When a PR touches PHP and JS, both the real and the fake summaries for node, eslint and phpunit got triggered. Since the fake summary is always green it meant one could merge a PR that in the end had red CI on the non-fake summary.

Now the fake summaries are skipped, if any of the files still matches the pattern, instead of being executed whenever a single file did not match it.

---

### How did I notice

* This PR was set to automerge https://github.com/nextcloud/spreed/pull/10317 and merged with red CI

### Reason

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-excluding-paths

> When all the path names match patterns in paths-ignore, the workflow will not run. If any path names do not match patterns in paths-ignore, even if some path names match the patterns, the workflow will run.

There was some file that didn't match, so that's why the "when unrelated" ran... that's .... problematic.

"When unrelated" should only run, when none of the matching files was touched.

### Debug

Debugged this in https://github.com/nickvergessen/turbo-lamp/pulls

* In https://github.com/nickvergessen/turbo-lamp/pull/3 you can see the "fake summary" being skipped successfully.

### Possible improvements

Read the .yml file and use the `on.pull_request.paths-ignore` instead of duplicating the paths into the env. But I was unable to achieve that and this solution here works good enough for now.

### CODE REVIEW

**Review by skjnldsv — APPROVED** [review]

**Review by max-nextcloud — APPROVED** [review]
Thanks for diving into this! 
Fixes #124

**Comment by nickvergessen** [comment]
So, this or you want to port your logic @kesselb

**Comment by kesselb** [comment]
Tough question :cold_sweat: 

I used the GitHub action dorny/paths-filter to run phpunit only for php* changes at https://github.com/nextcloud/server/pull/39734/files.

One advantage is that we only have one workflow. I find that easier to understand and maintain. 

The disadvantage is that the workflow itself is more complex because we have the additional job to evaluating the changed files and another check for the summary step. 

What do you think @skjnldsv @max-nextcloud @ChristophWurst?

File: workflow-templates/lint-eslint.yml [hunk 1]
```diff
@@ -31,10 +19,31 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  changes:
```
Comment (kesselb) [inline workflow-templates/lint-eslint.yml:22]:
```
outputs:
      src: ${{ steps.filter.outputs.src }}
```

Note: filter is the id for the step. I'm fine with changes but used a different id to not confuse it with the job id which is also changes. The outputs section is necessary if we want to use the filter result in a different job within the same workflow. Within the same job it's possible to access the data by the step id.

Comment (kesselb) [reply workflow-templates/lint-eslint.yml:22]: Ah, you just changed that for the spreed pr ;)

**Review by kesselb — APPROVED** [review]

**Pull request merged**